subprojects/core-api/src/main/java/org/gradle/api/IsolatedProject.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+
+/**
+ * An isolated view of {@link Project} that exposes only those properties that are safe to access from outside of
+ * <code>this</code> project, from the perspective of isolated projects.
+ * @since 8.7
+ */
+@Incubating
+public interface IsolatedProject {
```

Comment:
What about version and group to support project dependencies? AFAIK project dependencies aren't possible right now due getting an error when creating a ProjectDependency.

Reply:
Unfortunately `version` and `group` are mutable state.